### DISCUSSION TITLE
Adding NVIDIA App

### DISCUSSION THREAD

**Discussion by GreenMan36** [microsoft/winget-pkgs#200910] [General] [answered]
The [NVIDIA app](https://www.nvidia.com/en-us/software/nvidia-app/), which is the official successor to GeForce Experience, is currently unable to be added to the winget repository due to hardware validation issues. However, there are several important considerations that warrant discussion:

## Current Situation
- Multiple attempts to add the NVIDIA app have been blocked
  - #190775
  - #179208
  - #198796
  - #172329
- Nearly all attempts were marked with "Hardware" and "Blocking-Issue" labels
- The primary blocker is issue #199504 "Support hardware requirements" which is marked as "Not Planned"
- [There is also no static url to the latest version](https://github.com/microsoft/winget-pkgs/issues/140696#issuecomment-2492764409)

## Key Contradictions
1. GeForce Experience is already in the winget repository despite having identical hardware requirements. (Maybe I'm missing something here?)
3. NVIDIA app is officially replacing GeForce Experience, making it the recommended upgrade path for existing users.
4. The current situation creates an inconsistency where the deprecated software is available but its replacement isn't.

## Problems This Creates
1. Users can't easily install the new NVIDIA app through winget
2. GeForce Experience users looking to upgrade through winget have no official path forward
3. This effectively maintains deprecated software in the repository while blocking its replacement

### 1. Remove Legacy App
- GeForce Experience is officially deprecated by NVIDIA
- Maintaining it in the repository could mislead users into installing outdated software
- If hardware requirements are a blocker, this should apply to both the old and new app consistently

### 2. Add NVIDIA App Despite Hardware Validation Limitations
- Accept the app without hardware validation, consistent with how GeForce Experience was handled
- Let users handle hardware compatibility, as they already do with GeForce Experience
- This maintains feature parity with other package managers and installation methods

### 3. Create an Upgrade Path
Options include:
- Redirect Nvidia.GeForceExperience to the new NVIDIA app package
- Add deprecation warnings to GeForce Experience manifest
- Update GeForce Experience description to inform users about the replacement

Any combination of these proposals could be implemented. The key is to ensure we don't maintain a deprecated application while blocking its official replacement.

Reference: Original package request #140696

Would love to hear the community's thoughts on how we can move forward with any of these proposals. Or getting a clear cut answer on why this will never be the case and what will happen to the legacy app.

**Comment by GreenMan36** [comment]
I also have a couple of questions:

1. Hardware Requirements
- How was GeForce Experience (and others) initially added despite hardware requirements?
- Are there other examples of hardware-dependent software in winget we should take into consideration?
- What makes the NVIDIA app validation different from GeForce Experience?

2. Package Management
- How are other package managers handling the NVIDIA app?
- What approaches could we take for maintaining both apps during a transition period?
- How do we handle deprecated software in the repository?

3. User Experience
- How are users currently handling this limitation?
- What would be the most user-friendly approach to this transition?

**Reply by Trenly** [reply, marked as answer]
Geforce Experience will install and run even if there is not a physical Nvidia card in the system, which is how it is able to pass the validation pipelines. If Nvidia App has an installer switch that would allow it to install even if there is no Nvidia hardware, then we could add it.

The primary issue here is that all apps must go through the validation pipelines and be assesed that they install, run, and don't contain malware. I believe that other package managers like Scoop and Chocolatey have significantly different processes for testing these; if I'm not mistaken, the enterprise nature of Microsoft leads to a need for more strict scrutiny of packages.

As such, until Nvidia App can successfully pass the validation pipelines, there isn’t a whole lot that we can do
